### ISSUE TITLE
[BUG] destinationSet.removeAllExcept doesn't remove destinations if no match is found

### ISSUE THREAD

**Issue by mhokanson** [nextgenhealthcare/connect#5875]
**Describe the bug**
When using `destinationSet.removeAllExcept(<destination name>)` all destinations are enabled if no destinations match the supplied parameter.

This can lead to over-transmission of messages if there is a typo or some other issue in the `destinationSet.removeAllExcept(<destination name>)`, leading to unintended PHI disclosures.

**To Reproduce**
Setup steps (if required). Example:
1. Setup table in database with this schema
2. Add rows '...'

1. Set up multiple destinations
2. In a source filter or transformer use `destinationSet.removeAllExcept(<destination name>)` with a typo in the `<destination name>` value 
3. Process a message which would have triggered destination removal if the typo weren't present
4. Observe that no destinations were removed

**Expected behavior**
All destinations would be removed except the supplied destination. If no match is found all destinations should be removed (remove all except the one that wasn't found).

**Actual behavior**
When using `destinationSet.removeAllExcept(<destination>)` all destinations are enabled if no destinations match the supplied parameter.

**Environment (please complete the following information):**
 - OS: Windows 10, CentOS (version unknown)
 - Tested on Connect v3.10.1 (CentOS) and Connect v3.12.0 (Win 10)

**Workaround(s)**
Catch the output of destinationSet.removeAllExcept and perform a destinationSet.removeAll() if no destinations were removed.

**Comment by pacmano1** [comment]
On 4.3, it does not appear to work that way.

In the source transformer:

```JS
destinationSet.removeAllExcept(['ad']);
```

Send a message in and all three destinations are in fact removed.

The three destinations are named

1. `Destination 1`
2. `Destination 2`
3. `Destination 3`

**Comment by pacmano1** [comment]
I suspect you are doing the below, i.e. you are not passing an array.

```JS
destinationSet.removeAllExcept('ad');
```

**Comment by mhokanson** [comment]
Indeed I am (I copied what my predecessor had done).

Does 4.3 remove all three destinations if you pass a string instead of an array? Passing a string seems to remove all but the desired channels in my tests IF there is a match.

**Comment by pacmano1** [comment]
Blaming the last person, lol.

The call needs an array, not a string so 4.3 does the same thing.   I suppose you might change/update your issue to:

`destinationSet.removeAllExcpept() should throw an error if an array is not passed`

**Comment by mhokanson** [comment]
I'll just log a new issue for that. Thanks!

**Comment by tonygermano** [comment]
> The call needs an array, not a string so 4.3 does the same thing. I suppose you might change/update your issue to:
> 
> `destinationSet.removeAllExcpept() should throw an error if an array is not passed`

This is not true. There are two versions of DestinationSet.removeAllExcept. One takes an `Object` and the other takes a `Collection<Object>`. A JS array satisfies the second one, but a plain string should work as well.

**Comment by tonygermano** [comment]
I'm able to confirm the bug. It only occurs when you are using a destination name which doesn't exist. If you pass it a number for a metadata id which doesn't exist, then it will correctly remove all destinations.

**Comment by tonygermano** [comment]
The problem is in this method:

https://github.com/nextgenhealthcare/connect/blob/010f2a4dc636ad1c24b20a7fd1f625c5f6b91a39/server/src/com/mirth/connect/server/userutil/DestinationSet.java#L100-L110

Lines 104-106 should be changed to something like:
```java
         final Set<Integer> set = (metaDataId != null) ? Collections.singleton(metaDataId) : Collections.emptySet();
         return metaDataIds.retainAll(set); 
```
